### ISSUE TITLE
How to add product_appearance attribute to the basic information cluster (CON-1248)

### ISSUE THREAD

**Issue by FMistrorigo** [espressif/esp-matter#1002]
Hi, 
I wanted to add to my product the informations about color and finish, I found out this is possible in the attribute "product_appearance" of the cluster "basic information".
In my case the root node is created with the cluster basic information but the only attributes are node_label and cluster_revision, I know that using the function create_product_appearance it's possible to create the attribute.
![image](https://github.com/user-attachments/assets/15da84a1-8e42-434d-98ae-18123725c988)

I couldn't fins an example where this function is used, and in the documentation is not clear what parameters I have to provide in "value", "length" and "count".

What parameters do I have to provide if I want to display color: white and finish: satin when in the chip tool i use the command "./chip_tool basicinformation read product-appearance 1 0"?

Thanks

**Comment by shripad621git** [comment]
@FMistrorigo , thanks for reporting the issue. The product appearance attribute can be set by creating a factory_partiton using `esp-matter-mfg-tool`. It is read using the DeviceInstanceInfoProvider implementation. We will add the support for the attribute soon.

**Comment by shripad621git** [comment]
@FMistrorigo ,the change is now present in connectedhomeip [here](https://github.com/project-chip/connectedhomeip/pull/34363). 
The `product-appearanace` attribute is read using the platform specific [DeviceInstanceInfoProvider](https://github.com/project-chip/connectedhomeip/blob/5e3127f5ac61e13c572a968199280d90a9c19dce/src/platform/ESP32/ESP32FactoryDataProvider.cpp#L248)  implementation 
Till we update the submodule, you can apply do these changes accordingly in your connectedhomeip submodule and try out the product-appearance attribute in esp-matter.
For trying this out, you need to create factory partition containing the product-finish and product-color attributes.
For e.g:
```
  ./generate_esp32_chip_factory_bin.py -d 3434 -p 99663300 \
                                    --product-name ESP-lighting-app --product-id 0x8001 \
                                    --vendor-name Test-vendor --vendor-id 0xFFF2 \
                                    --hw-ver 1 --hw-ver-str DevKit  --product-finish polished --product-color navy
```
You can try out the factory script present [here](https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/generate_esp32_chip_factory_bin.py) with the changes in the above PR for generating the factory partition. 
You will have to enable the following options in the menuconfig for testing out the prodcut-appearance attribute.
```
CONFIG_ENABLE_ESP32_FACTORY_DATA_PROVIDER=y
CONFIG_ENABLE_ESP32_DEVICE_INSTANCE_INFO_PROVIDER=y
```
For more information on factory data, you can refer this [link](https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/esp32/factory_data.md).

**Comment by shripad621git** [comment]
@FMistrorigo , any updates on the issue.

**Comment by FMistrorigo** [comment]
Hi @shripad621git,

I didn't want to create a Factory partition only for these attributes, for the Device Instance Info Provider I use a custom one and I tried to implement the two functions in this way, so when the get functions were called the output was what I wanted, but it didn't work.
![image](https://github.com/user-attachments/assets/8e954266-fcc8-4237-bbae-f04ac38102f7)

I decided to go back to the function create_product_appearance and did this:
![image](https://github.com/user-attachments/assets/d293c746-0340-44b2-ac7f-d036e9f63f46)
(0x02 is Satin, and 0x0E is White)

Everything was good in the chip tool was working so I left it like this.


Tell me if ther's any problem in what I did.
Thanks.
